service/geopmdpy/system_files.py

```diff
@@ -723,6 +731,51 @@ def remove_batch_server(self, client_pid):
             sys.stderr.write(f'Warning: {write_fifo_path} file was left over, deleting it now.\n')
             os.unlink(write_fifo_path)
 
+    def start_profile(self, client_pid, profile_name):
+        self.check_client_active(client_pid, 'start_profile')
+        uid, gid = self._pid_info(client_pid)
+        if len(self._profiles) == 0:
+            size = 64 * os.cpu_count()
+            shmem.create_prof('status', size, client_pid, uid, gid)
+        if profile_name in self._profiles:
+            self._profiles[profile_name].add(client_pid)
+        else:
+            self._profiles[profile_name] = {client_pid}
+        self._sessions[client_pid]['profile_name'] = profile_name
+        size = 49192
+        shmem.create_prof('record-log', size, client_pid, uid, gid)
+        self._update_session_file(client_pid)
+
+    def stop_profile(self, client_pid, region_names):
+        self.check_client_active(client_pid, 'stop_profile')
+        try:
+            profile_name = self._sessions[client_pid].pop('profile_name')
+        except KeyError:
+            raise RuntimeError(f'Client PID {client_pid} requested to stop profiling, but it had not been started.')
+        # TODO: store region names in file in /run/geopm-service to enable clean restart
+        if profile_name in self._region_names:
+            self._region_names[profile_name].update(region_names)
+        else:
+            self._region_names[profile_name] = set(region_names)
+        self._profiles[profile_name].remove(client_pid)
+        uid, gid = self._pid_info(client_pid)
+        os.unlink(shmem.path_prof('record-log', client_pid, uid, gid))
+        if len(self._profiles[profile_name]) == 0:
+            self._profiles.pop(profile_name)
+        if len(self._profiles) == 0:
+            uid, gid = self._pid_info(client_pid)
+            os.unlink(shmem.path_prof('status', client_pid, uid, gid))
+        self._update_session_file(client_pid)
+
+    def get_profile_pids(self, profile_name):
+        return self._profiles.get(profile_name)
+
+    def get_profile_region_names(self, profile_name):
+        result = None
+        if profile_name in self._region_names:
+            result = self._region_names.pop(profile_name)
```

Comment:
We need some mechanism for clean up of the resources associated with the application run, and since we currently have only one consumer of this information (the controller), I put the clean up in the function the controller calls once at the end.  This is a nasty side effect of this method, and we should have a better story for cleaning up.

Reply:
So maybe it's just a matter of changing the method's name, for example, something like `pop_profile_region_names()`